dflib/src/test/java/org/dflib/ql/ExpParamsTest.java

```diff
@@ -0,0 +1,109 @@
+        Exp<?> exp = parseExp(text, args);
+        assertEquals(expected, exp);
+    }
+
+    static Stream<Arguments> parameterizedExpression() {
+        return Stream.of(
+                arguments(
+                        "?",
+                        new Object[]{true},
+                        $boolVal(true)
+                ),
+
+                arguments(
+                        "int(1) > ?",
+                        new Object[]{1L},
+                        $int(1).gt($longVal(1L))
+                ),
+                arguments(
+                        "str(b) != ?",
+                        new Object[]{"abc"},
+                        $str("b").ne($strVal("abc"))
+                ),
+                arguments(
+                        "bool(c) = ?",
+                        new Object[]{true},
+                        $bool("c").eq($boolVal(true))
+                ),
+
+                arguments(
+                        "date(c) = ?",
+                        new Object[]{LocalDate.of(1905, 4, 3)},
+                        $date("c").eq($dateVal(LocalDate.of(1905, 4, 3)))
+                ),
+                arguments(
+                        "time(c) = ?",
+                        new Object[]{LocalTime.of(23, 4, 3)},
+                        $time("c").eq($timeVal(LocalTime.of(23, 4, 3)))
+                ),
+                arguments(
+                        "dateTime(c) = ?",
+                        new Object[]{LocalDateTime.of(
+                                LocalDate.of(1905, 4, 3),
+                                LocalTime.of(23, 4, 3)
+                        )},
+                        $dateTime("c").eq($dateTimeVal(LocalDateTime.of(
+                                LocalDate.of(1905, 4, 3),
+                                LocalTime.of(23, 4, 3))
+                        ))
+                ),
+                arguments(
+                        "offsetDateTime(c) = ?",
+                        new Object[]{OffsetDateTime.of(
+                                LocalDate.of(1905, 4, 3),
+                                LocalTime.of(23, 4, 3),
+                                ZoneOffset.ofHours(1)
+                        )},
+                        $offsetDateTime("c").eq($offsetDateTimeVal(OffsetDateTime.of(
+                                LocalDate.of(1905, 4, 3),
+                                LocalTime.of(23, 4, 3),
+                                ZoneOffset.ofHours(1))
+                        ))
+                ),
+
+                arguments(
+                        "int(1) in (?)",
+                        new Object[]{List.of(1, 2, 3)},
+                        $int(1).in(1, 2, 3)
+                ),
+                arguments(
+                        "str(1) in (?)",
+                        new Object[]{List.of("a", "b", "c")},
+                        $str(1).in("a", "b", "c")
+                ),
+                arguments(
```

Comment:
We kinda need to distinguish between a list of values for the `in` clause vs individual values in that list. The below example is visually ambiguous. So what if we follow a rule like this:

```java
// no parenthesis - expect a list
"a in ?"

// in parenthesis - just a scalar value
"a in (?)"
"a in (?, "b", ?)"
```